evalml/data_checks/target_leakage_data_check.py

```diff
@@ -13,23 +13,49 @@
 class TargetLeakageDataCheck(DataCheck):
     """Check if any of the features are highly correlated with the target."""
 
-    def __init__(self, pct_corr_threshold=0.95):
+    def __init__(self, pct_corr_threshold=0.95, pearson_corr=False):
```

Comment:
A little nudge towards explicitness: perhaps we might consider a conditional on a string with the positional arg being named something like "corr_check_method" and the default being "pearson" with "mutual_information" being an option?  I think it's a little clear then what's happening if not Pearson correlation.

Reply:
@chukarsten thanks for that pointer! It definitely would be better to be more explicit here. To make it easier for the user, I'll leave the input as is, but I'll update the documentation to let users know that it uses mutual information as a default, and Pearson correlation otherwise. Having a boolean arg seems simpler for the user versus a string arg.